packages/tanstack-react-query/test/queryKeyable.test.tsx

```diff
@@ -141,8 +141,8 @@ describe('get queryFilter', () => {
         {
           predicate(query) {
             const data = query.setData('__result');
-            assertType<'__result' | undefined>(data);
-            assertType<TRPCQueryKey>(query.queryKey);
+            assertType<unknown>(data);
```

Comment:
This is a downgrade?

Reply:
so TRQ removed type safety from the filters and predicates because it was incorrect, the only instance you would ever use the predicate is where you don't know the type because you're fuzzy searching so it was decided to move back to unknoqn

Reply:
See https://github.com/TanStack/query/issues/8768